tools/schema_loader.cc

```diff
@@ -161,12 +163,44 @@ class data_dictionary_impl : public data_dictionary::impl {
         static const std::vector<view_ptr> empty;
         return empty;
     }
-    virtual sstring get_available_index_name(data_dictionary::database db, std::string_view ks_name, std::string_view table_name,
+    virtual sstring get_available_index_name(data_dictionary::database db, std::string_view ks_name, std::string_view cf_name,
             std::optional<sstring> index_name_root) const override {
-        throw std::bad_function_call();
+        auto has_schema = [&] (std::string_view ks_name, std::string_view table_name) {
+            const auto& tables = unwrap(db).tables;
+            return std::find_if(tables.begin(), tables.end(), [&] (const table& t) {
+                return t.schema->ks_name() == ks_name && t.schema->cf_name() == table_name;
+            }) != tables.end();
+        };
+        // Copied from replica::database::get_available_index_name()
+        auto existing_names = existing_index_names(db, ks_name);
+        auto base_name = index_metadata::get_default_index_name(sstring(cf_name.data()), index_name_root);
+        sstring accepted_name = base_name;
+        int i = 0;
+        auto name_accepted = [&] {
+            auto index_table_name = secondary_index::index_table_name(accepted_name);
+            return !has_schema(ks_name, index_table_name) && !existing_names.contains(accepted_name);
+        };
+        while (!name_accepted()) {
+            accepted_name = base_name + "_" + std::to_string(++i);
+        }
+        return accepted_name;
```

Comment:
Neither works:
1. I cannot actually execute these statements, I do not have a functioning database in the tool, in fact I try very hard to avoid starting any of the database services in the tool. They might start to write stuff to disk, or even worse -- modify existing system tables of the examined ScyllaDB node.
2. The existing function is a method of `replica::database`. I will look into maybe extracting that logic into a standalone function, called by both the db and the schema loader.

Reply:
I extracted the code into a common standalone function in v2.